main.py

```diff
@@ -128,13 +153,13 @@ async def on_message(message):
                 and len(split_message[1]) <= 254:
                 email = split_message[1]
                 # Validate email
-                if is_valid_email(email): #TODO and not_validated(member):
+                if is_valid_email(email) and await is_free_email(member, email):
                     print("Valid email used: " + email)
                     gen = CodeGenerator(code_length=CODE_LENGTH)
                     code = gen.generate()
                     print("Code generated:" +  code)
                     emailService = EmailService(EMAIL_USERNAME,EMAIL_PASSWORD)
-                    status = emailService.sendmail(receiver=email, 
+                    status = emailService.sendmail(receiver=email,
```

Comment:
We have to change the text here as well, maybe it should get defined as a constant or env file